internal/controller/nonadminbackup_controller.go

```diff
@@ -348,8 +354,8 @@ func updateNonAdminBackupVeleroBackupReference(status *nacv1alpha1.NonAdminBacku
 // updateNonAdminBackupVeleroBackupStatus sets the VeleroBackup status field in NonAdminBackup object status and returns true
 // if the VeleroBackup fields are changed by this call.
 func updateNonAdminBackupVeleroBackupStatus(status *nacv1alpha1.NonAdminBackupStatus, veleroBackup *velerov1.Backup) bool {
```

Comment:
Check both the status to be not nil ? (Most likely need a nil check for veleroBackup.status )